engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/dict/SpelDictTyper.scala

```diff
@@ -0,0 +1,67 @@
+package pl.touk.nussknacker.engine.dict
+
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.ValidatedNel
+import org.springframework.expression.spel.SpelNode
+import org.springframework.expression.spel.ast.{Indexer, PropertyOrFieldReference, StringLiteral}
+import pl.touk.nussknacker.engine.api.expression.ExpressionParseError
+import pl.touk.nussknacker.engine.api.typed.typing.{DynamicTypedDict, StaticTypedDict, Typed, TypedDict, TypingResult}
+import pl.touk.nussknacker.engine.spel.ast
+
+/**
+  * It handle typing process off SpEL AST in places where typer referencing to TypedDict type.
+  */
+trait SpelDictTyper {
+
+  def typeDictValue(dict: TypedDict, node: SpelNode): ValidatedNel[ExpressionParseError, TypingResult]
+
+}
+
+trait BaseDictTyper extends SpelDictTyper {
+
+  import ast.SpelAst._
+
+  override def typeDictValue(dict: TypedDict, node: SpelNode): ValidatedNel[ExpressionParseError, dict.ValueType]  = {
+    node match {
+      case _: Indexer =>
+        node.children match {
+          case (str: StringLiteral) :: Nil  =>
+            val key = str.getLiteralValue.getValue.toString
+            findKey(dict, key)
+          case _ =>
+            Invalid(ExpressionParseError(s"Illegal spel construction: ${node.toStringAST}. Dict should be indexed by a single key")).toValidatedNel
+        }
+      case pf: PropertyOrFieldReference  =>
+        findKey(dict, pf.getName)
+    }
+  }
+
+  private def findKey(dict: TypedDict, key: String) = {
```

Comment:
maybe we should consider moving this logic to substitution? And let Typer focus solely on Typing?

Reply:
Yeah I will do kind of such stuff when I will resolve TODO in TypedDict. It is connected.